### PR TITLE
feat: [PL-30423]: Added aggregated API for cards on streaming destination page

### PR DESCRIPTION
**Pull request by nishantsaini20**
## Describe your changes
Streaming Destination page requires aggregated API to get info for cards on the page. Added API for the same.

Testing:
- Tested in local env.
- Added UTs

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)

### CODE REVIEW

**Review by piyush-harness — DISMISSED** [review]
> Reviewed 37 of 38 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: 38 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 38 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: 37 file(s) to review. 1 of 2 approvals obtained.

File: platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -119,8 +119,34 @@ paths:
...
+      parameters:
+        - $ref: '#/components/parameters/AccountHeader'
 components:
   schemas:
+    StreamingDestinationStatus:
+      title: StreamingDestinationStatus
+      x-stoplight:
+        id: 6k8zse7tcg5y0
+      type: string
+      x-extension-1: null
```
Comment (mankrit-singh) [inline platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml:144]: nit, what is this extension?

**Review by reetika04 — APPROVED** [review]
> Reviewed 30 of 38 files at r1, 1 of 1 files at r2, 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: 7 file(s) to review.

File: platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (nishantsaini20) [reply platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml:144]: Looks like added by stoplight studio.

**Review by reetika04 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: 7 file(s) to review. 1. unresolved discussions.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 30 of 38 files at r1, 6 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: 1 file(s) to review.

**Review by piyush-harness — APPROVED** [review]
> Reviewed 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43175)* status: :shipit: complete! ✓

**Pull request merged**